client/src/components/DataFiles/DataFilesModals/DataFilesCompressModal.js

```diff
@@ -0,0 +1,197 @@
+                name="filetype"
+                bsSize="sm"
+                onChange={handleSelectChange}
+              >
+                <option value="zip">zip</option>
+                <option value="tar.gz">tar.gz</option>
+              </Input>
+            </FormGroup>
+          </Form>
+        );
+      }}
+    </Formik>
+  );
+};
+DataFilesCompressForm.propTypes = {
+  first: PropTypes.string.isRequired,
+  formRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({ current: PropTypes.instanceOf(Element) })
+  ]).isRequired
+};
+
+const DataFilesCompressModal = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const status = useSelector(
+    state => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const params = useSelector(
+    state => state.files.params.FilesListing,
+    shallowEqual
+  );
+
+  const isOpen = useSelector(state => state.files.modals.compress);
+  const selectedFiles = useSelector(
+    ({ files: { selected, listing } }) =>
+      selected.FilesListing.map(i => ({
+        ...listing.FilesListing[i]
+      })),
+    shallowEqual
+  );
+  const selected = useMemo(() => selectedFiles, [isOpen]);
+  const formRef = React.useRef();
+  const toggle = () =>
+    dispatch({
+      type: 'DATA_FILES_TOGGLE_MODAL',
+      payload: { operation: 'compress', props: {} }
+    });
+
+  const onOpened = () => {
+    dispatch({
+      type: 'FETCH_FILES_MODAL',
+      payload: { ...params, section: 'modal' }
+    });
+  };
+
+  const onClosed = () => {
+    dispatch({ type: 'DATA_FILES_MODAL_CLOSE' });
+    if (isString(status)) {
+      dispatch({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: {}, operation: 'compress' }
+      });
+      history.push(location.pathname);
+    }
+  };
+
+  const compressCallback = () => {
+    const { filename } = formRef.current.values;
+    dispatch({
+      type: 'DATA_FILES_COMPRESS',
+      payload: { filename, files: selected }
+    });
+  };
+
+  let buttonIcon;
+  if (status === 'RUNNING') {
+    buttonIcon = <LoadingSpinner placement="inline" />;
+  } else if (status === 'ERROR') {
+    buttonIcon = <Icon name="alert" />;
+  } else {
+    buttonIcon = null;
+  }
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onOpened={onOpened}
+      onClosed={onClosed}
+      toggle={toggle}
+      className="dataFilesModal"
+    >
+      <ModalHeader toggle={toggle}>Compress Files</ModalHeader>
+      <ModalBody>
+        <DataFilesCompressForm
+          formRef={formRef}
+          first={(selectedFiles[0] && selectedFiles[0].name) || ''}
+        />
+        <p>
+          A job to compress your files will be submitted on your behalf. You can
+          check the status of this job on your Dashboard, and your compressed
+          file will appear in this directory.
+        </p>
+        {status === 'SUCCESS' && (
+          <span style={{ color: 'green' }}>
```

Comment:
what about using _common/Message here?  (same question applies for Extract modal)

Reply:
I was working on changes to the push keys modal and this looks like something we should use the `<InlineMessage>` notification for. Could double check with @tacc-wbomar but this success/failure notification would look best in line with the submit button like shown in the image below. We should also remove the "Close" button from this compress modal:
![image](https://user-images.githubusercontent.com/29575979/102284552-f69a0580-3ef9-11eb-93e0-feb162574608.png)

Reply:
Yes. Great suggestion. Thanks.

That is similar to the [UI defined for success message](https://jira.tacc.utexas.edu/secure/attachment/14723/Late%20Design%20-%20Action%20Success.png) via [FP-740](https://jira.tacc.utexas.edu/browse/FP-740).

#### `<InlineMessage>` Alignment & Size

Specific placement details of `<InlineMessage>`s and `<SectionMessage>`s has not been fully defined, yet.

- **✓** __Left-align with normal text.__
    - _Easiest Non-Committal Option_
- **?** Right-align with small text.
    - _Complications exist:_
        - Quick & Consistent: `<InlineMessage tagName="small">` (but this prop is only available after PR #277)
        - Alternative: Use CSS (module) class names to reduce the size._

Reply:
**Update**: ✓ Right-align with normal text.